nautobot/docs/user-guide/core-data-model/users/savedview.md

```diff
@@ -0,0 +1,46 @@
+# Saved Views
+
++++ 2.3.0
+
+Saved Views give users the ability to save multiple configurations of list views, including table columns, filtering, pagination, and sorting, for ease of later use and reuse. Check out this [user guide](../../feature-guides/saved-views.md) on how to use saved views.
+
+The essential attributes for a Saved View are `owner`, `name`, `view`, `is_shared` and `is_global_default`. The `owner` attribute links to the user account active during the creation of the Saved View. The `view` attribute refers to the specific list view from which the Saved View is derived, such as `dcim:device_list` or `circuits:circuit_list`.
+The `is_shared` attribute dictates whether the saved view will be public or private. The `is_global_default` attribute dictates whether all users trying to access the object list view indicated by the value of the `view` attribute will be redirected to the current saved view or not.
```

Comment:
It occurs to me (somewhat belatedly) that `is_global_default` is a proper database field, but per-user defaults are stored in the user `config_data` JSON, that is to say they are denormalized and are not actual foreign keys. I wonder whether it would be better design wise if we had a proper many-to-many mapping between each user and their per-user default views. It'd be more self-consistent, and would let us use attributes like `on_delete=CASCADE` so that a user would never have "stale" saved view defaults. What do you think?

Reply:
Yeah that does sound like a better design, I will refactor it.

Reply:
Okay, one thing I noticed immediately when I tried to refactor this feature is that we need extra validation to make sure that there are no saved views with the same `view` attribute mapped to the same user, which was not needed in the original implementation.

Reply:
Should be doable with a `unique_together` constraint on the through table, but agreed.

Reply:
Hmmm I wonder if `unique_together` supports indirect fields. Something like`unique_together = [("user", "saved_view__name"),]`. Maybe we can store `"saved_view__name"` in a direct field of the through table like `view_name`